torch/_inductor/runtime/benchmarking.py

```diff
@@ -0,0 +1,944 @@
+                    x.grad = None
+            # we clear the L2 cache before each run
+            cache.zero_()
+            # record time of `fn`
+            start_event[i].record()
+            fn()
+            end_event[i].record()
+        # Record clocks
+        torch.cuda.synchronize()
+        times = torch.tensor(
+            [s.elapsed_time(e) for s, e in zip(start_event, end_event)],
+            dtype=torch.float,
+        )
+        if quantiles is not None:
+            ret = torch.quantile(
+                times, torch.tensor(quantiles, dtype=torch.float)
+            ).tolist()
+            return ret[0]
+        return getattr(torch, return_mode)(times).item()
+
+    @time_and_log
+    @maybe_fallback_to_original_benchmarking("original_do_bench_many_gpu")
+    def benchmark_many_gpu(
+        self: Self,
+        callables: List[Callable[[], Any]],
+        estimation_iters: int = 5,
+        memory_warmup_iters: int = 100,
+        benchmark_iters: int = 100,
+        max_benchmark_duration_ms: int = 25,
+        ranking_key: Optional[str] = None,
+        pruning_key: Optional[str] = None,
+        pruning_factor: float = 1.1,
+        pruning_limit: int = 5,
+        **kwargs: Any,
+    ) -> List[float]:
+        # we don't want any outside errors propagating into benchmarking
+        torch.cuda.synchronize()
+
+        # initialize here, avoids double buffer allocation
+        self.cpu_launch_overhead_ms_and_gpu_time_ms_per_gpu_cache_clear
+
+        # this will fail the entire group if any of the callables fail,
+        # in the case that users directly call this method they should
+        # handle this case on their end. in the case that we get here
+        # from a grouping of lazy benchmarks, we should never have failures
+        # since they will have already been checked previously
+        for _callable in callables:
+            _callable()
+        torch.cuda.synchronize()
+
+        buffer = torch.empty(self.L2_cache_size // 4, dtype=torch.int, device="cuda")
+        buffer.zero_()
+
+        interleaved_event_pairs = self.get_interleaved_event_pairs(
+            len(callables), estimation_iters
+        )
+        torch.cuda._sleep(int(1 / self.gpu_time_ms_per_gpu_clock_cycle))
+        start_time_s = perf_counter()
+        for event_pairs in interleaved_event_pairs:
+            for _callable, (start_event, end_event) in zip(callables, event_pairs):
+                buffer.zero_()
+                start_event.record()
+                _callable()
+                end_event.record()
+        cpu_launch_overhead_ms_per_iter = (
+            (perf_counter() - start_time_s) * MILLISECONDS_PER_SECOND
+        ) / estimation_iters
+        torch.cuda.synchronize()
+        estimated_timings_ms = self.get_interleaved_min_timings_ms(
+            interleaved_event_pairs
+        )
+
+        if ranking_key is not None:
+            if should_enable_early_ranking():
+                log.debug(
```

Comment:
Just for my own understanding - enable_early_ranking is the default today ? And we might want to fallback to this if there is an error with the new logic ? If so, `should_enable` is not api i would expect.

Reply:
`enable_early_ranking` defaults to True in OSS, False internal. Would you expect `should_disable`?